src/DynamoCore/Engine/CodeGeneration/CompilingEventArgs.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace Dynamo.Engine.CodeGeneration
+{
+    /// <summary>
+    /// This event is triggerred when compiling a NodeModel to AST nodes.
+    /// </summary>
+    public class CompilingEventArgs : EventArgs
+    {
+        /// Construct ASTBuildingEventArgs with NodeModel.
+        public CompilingEventArgs(Guid node)
+        {
+            Node = node;
+        }
+
+        /// <summary>
+        /// Guid of NodeModel that is being compiled to AST.
+        /// </summary>
+        public Guid Node { get; private set; }
```

Comment:
this should definitely be called nodeId or something besides Node

Reply:
I agree, obsoleted and NodeId property added.